File: java/data/src/main/java/com/youknow/data/source/remote/api/model/NowPlayingResp.java
package com.youknow.data.source.remote.api.model;

import com.google.gson.annotations.SerializedName;

import com.youknow.domain.model.SimpleMovie;

import java.util.ArrayList;
import java.util.List;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class NowPlayingResp {
    private int page;
    private List<Result> results;
    private Dates dates;
    private int totalPages;
    private int totalResults;

    public static List<SimpleMovie> mapToDomain(NowPlayingResp nowPlayingResp) {
        System.out.println("[Y.M.] mapToDomain - " + nowPlayingResp);
        List<SimpleMovie> simpleMovieList = new ArrayList<>();
        for (Result result : nowPlayingResp.results) {
            simpleMovieList.add(new SimpleMovie(result.getPosterPath(), result.isAdult(), result.getReleaseDate(), result.getId(), result.getTitle(), result.getVoteAverage()));
        }

        return simpleMovieList;
    }
}

@Data
@NoArgsConstructor
class Result {
    @SerializedName("poster_path")
    private String posterPath;
    private boolean adult;
    private String overview;
    @SerializedName("release_date")
    private String releaseDate;
    @SerializedName("genre_ids")
    private List<Integer> genreIds;
    private int id;
    @SerializedName("original_title")
    private String originalTitle;
    @SerializedName("original_language")
    private String originalLanguage;
    private String title;
    @SerializedName("backdrop_path")
    private String backdropPath;
    private float popularity;
    @SerializedName("vote_count")
    private int voteCount;
    private boolean video;
    @SerializedName("vote_average")
    private float voteAverage;
}

@Data
@NoArgsConstructor
class Dates {
    private String maximum;
    private String minimum;
}

File: kotlin-rx/data/src/main/java/com/youknow/data/datasource/remote/api/model/NowPlayingResp.kt
package com.youknow.data.datasource.remote.api.model

import com.google.gson.annotations.SerializedName
import com.youknow.domain.model.SimpleMovie

data class NowPlayingResp(
    val page: Int = 0,
    val results: List<Result> = listOf(),
    val dates: Dates = Dates(),
    val totalPages: Int = 0,
    val totalResults: Int = 0
)

data class Result(
    @SerializedName("poster_path")
    val posterPath: String = "",
    val adult: Boolean = false,
    val overview: String = "",
    @SerializedName("release_date")
    val releaseDate: String = "",
    @SerializedName("genre_ids")
    val genreIds: List<Int> = listOf(),
    val id: Int = 0,
    @SerializedName("original_title")
    val originalTitle: String = "",
    @SerializedName("original_language")
    val originalLanguage: String = "",
    val title: String = "",
    @SerializedName("backdrop_path")
    val backdropPath: String = "",
    val popularity: Float = 0f,
    @SerializedName("vote_count")
    val voteCount: Int = 0,
    val video: Boolean = false,
    @SerializedName("vote_average")
    val voteAverage: Float = 0f
)

data class Dates(
    val maximum: String = "",
    val minimum: String = ""
)

fun NowPlayingResp.mapToDomain(): List<SimpleMovie> = results.map {
    SimpleMovie(
        posterPath = it.posterPath,
        adult = it.adult,
        overview = it.overview,
        releaseDate = it.releaseDate,
        id = it.id,
        title = it.title,
        popularity = it.popularity,
        voteCount = it.voteCount,
        voteAverage = it.voteAverage
    )
}.toList()